one-ui/src/main/java/com/marklogic/hub/curation/services/FlowManagerService.java

```diff
@@ -0,0 +1,288 @@
+package com.marklogic.hub.curation.services;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.hub.FlowManager;
+import com.marklogic.hub.StepDefinitionManager;
+import com.marklogic.hub.error.DataHubProjectException;
+import com.marklogic.hub.oneui.exceptions.DataHubException;
+import com.marklogic.hub.flow.Flow;
+import com.marklogic.hub.flow.impl.FlowImpl;
+import com.marklogic.hub.oneui.models.StepModel;
+import com.marklogic.hub.scaffold.Scaffolding;
+import com.marklogic.hub.step.StepDefinition;
+import com.marklogic.hub.step.impl.Step;
+import com.marklogic.hub.util.json.JSONObject;
+import com.marklogic.hub.util.json.JSONUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+@Service
+public class FlowManagerService {
+
+    @Autowired
+    private FlowManager flowManager;
+
+    @Autowired
+    private StepDefinitionManager stepDefinitionManager;
+
+    @Autowired
+    private Scaffolding scaffolding;
+
+    public Flow createFlow(String flowJson, boolean isFlowPresent) {
```

Comment:
I would think the client shouldn't have to tell the service whether the flow exists yet or not - the service should know how to figure that out. The client just wants to say - here's my JSON, please save it.